### ISSUE TITLE
Diagnostics don't work when they include a stack trace

### ISSUE THREAD

**Issue by RSDuck** [saem/vscode-nim#15]
e.g. when a macro throws an error:
```nim
import macros
macro x(): untyped = error("Hello World")
x()
```
similar problem as https://github.com/saem/vscode-nim/pull/14 where a null value from the regex parsing silently slipped through without generating an exception in the original code, but doesn't in the translated code.

I'll soon make a PR.

**Comment by saem** [comment]
Apologies, I made a partial fix and the issue got closed.

The errors now show at least with the hacky guard I put in, but I think the whole parsing is not the right approach.

There seem to be at least a few major types of output when it comes to nim check:

1. prologue as in all the hints up front
2. `.` on the same line and possibly next line (not sure if it wraps via new line) if it's just making progress
3. an error, warning, etc... message on a single line
4. the start of a `template/generic ...` message
5. stack trace due to a macro error
6. stack trace because of a fatal error of nim check

In terms of "phases" of the various types of parsing, I think they're roughly:
* `1` happens and is done, the command outputs whatever else or crashes out, this is straight forward, but it should be easy to know we're done here as soon as we get any of the other outputs
* the next phase is handling `2` to `6`
* `2` & `3` are simple line by line and the only thing we need to worry about is if they're the file is in the workspace or not
* `4` to `5` are effectively a nested phases, as in while we're in them we want to aggregate information until an end point is indicated
** `4` the parsing for this should do a better job provide range information start and end for line and column information here because it knows more while it's sifting through the output
** `5` I think we want to keep track of the invocation, and the first & last in workspace position information (file, line, column) within the trace in order to best figure out a message -- basically we can have a few combinations of those three things being in and out of workspace code and the results should adapt for these
** `6` mostly need to return whatever partial success and then report any underlying errors as a separate matter

**Comment by RSDuck** [comment]
> the start of a template/generic ... message
stack trace due to a macro error
stack trace because of a fatal error of nim check

those seem to all work similar as they're come before the actual error and are in the standard error format except without severity. That's how I handled it in my PR.

**Comment by saem** [comment]
Fixed in #16.
